msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -49,15 +49,53 @@ public UIParent()
             CoreUIParent = new CoreUIParent();
         }
 
+#if iOS
+        public UIParent(bool useEmbeddedWebview)
+        {
+            CoreUIParent.UseEmbeddedWebview = useEmbeddedWebview;
+        }
+#endif
+
 #if ANDROID
+        private Activity Activity { get; set; }
+
+        private static readonly string[] _chromePackages =
+        {"com.android.chrome", "com.chrome.beta", "com.chrome.dev"};
 
         /// <summary>
         /// Initializes an instance for a provided activity.
         /// </summary>
         /// <param name="activity">parent activity for the call. REQUIRED.</param>
         public UIParent(Activity activity)
         {
-            CoreUIParent = new CoreUIParent(activity);
+            Activity = activity;
+            CoreUIParent = new CoreUIParent(Activity);
+        }
+
+        public UIParent(Activity activity, bool useEmbeddedWebview) :this(activity)
+        {
+            CoreUIParent.UseEmbeddedWebview = useEmbeddedWebview;
+        }
+
+        public static bool IsSystemWebviewAvailable()
+        {
+            PackageManager packageManager = Application.Context.PackageManager;
```

Comment:
>Application.Context.PackageManager; [](start = 44, length = 35)

to make this class unit testable this might be one of the things you inject instead of setting it from with-in the class?

Reply:
sounds good...will include a unit test in next PR for this. should have thought of that.